inst/pages/98_exercises.qmd

```diff
@@ -1158,62 +1155,128 @@ See the OMA book for examples.
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
 
-# Perform k-means clustering on the reduced dimensions of pcoa_bray
-pcoa_bray_clusters <- kmeans(reducedDim(pcoa_bray), centers = 3)$cluster
+# Load necessary libraries
+library(mia)
+library(SummarizedExperiment)
+library(vegan)
+library(bluster)
+library(scater)
 
-# Plot the reduced dimensions of pcoa_bray with clusters as colors
-plotReducedDim(pcoa_bray, colour_by = pcoa_bray_clusters)
+# Generate a sample SummarizedExperiment object
+data(GlobalPatterns)
+tse <- GlobalPatterns
 
-# Perform k-means clustering on the reduced dimensions of pca_aitchison
-pca_aitchison_clusters <- kmeans(reducedDim(pca_aitchison), centers = 3)$cluster
+# Perform MDS on the abundance data using scater::runMDS
+# Specify the correct assay to be used
+tse <- runMDS(tse, FUN = vegan::vegdist, method = "bray", assay.type = "counts")
 
-# Plot the reduced dimensions of pca_aitchison with clusters as colors
-plotReducedDim(pca_aitchison, colour_by = pca_aitchison_clusters)
+# Perform k-means clustering (k = 3) using the bluster package
+set.seed(123)
+clustering_result <- clusterRows(reducedDim(tse, "MDS"), BLUSPARAM=KmeansParam(centers=3))
+colData(tse)$cluster <- as.factor(clustering_result)
+
+# Visualize MDS with clusters
+plotReducedDim(tse, dimred = "MDS", colour_by = "cluster") +
+  labs(title = "MDS with k-means Clustering") +
+  theme_minimal() +
+  scale_color_manual(values = c("red", "blue", "green"))
 ```
-Where : 
-'reducedDim' is a function that extracts the reduced dimensionality representation of your data (e.g., from a PCA or 
-PCoA object).
-'plotReducedDim' is a function that can plot these reduced dimensions and has a 'colour_by argument' that accepts a vector of cluster assignments.
 
-#### 2. Heatmap with Annotations:
+### 2: Cluster Analysis with Heatmap Visualization
 
-* Annotate the heatmap with additional metadata (e.g., subject groups or sample types).
+- Hierarchical clustering of samples based on abundance data.
+- Visualizes the clustered data using a heatmap, with cluster information added to metadata.
 
 ```{r}
+#| label: heatmap-visualization-ex2
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
 # Load necessary libraries
 library(SummarizedExperiment)
-library(pheatmap)
+library(sechm)
+library(compositions)
 
-# Example data creation
-# Assuming enterotype is a SummarizedExperiment object with sample metadata
-# and assay_data is a matrix containing assay data
+# Generate a sample SummarizedExperiment object 
+data(GlobalPatterns)
+tse <- GlobalPatterns
+
+# CLR transformation of abundance data matrix
+clr_data <- as.matrix(transformAssay(tse, method = "clr", assay.type = "counts", pseudocount = TRUE))
+
+# Hierarchical clustering on samples
+clustering <- hclust(dist(clr_data, method = "euclidean"), method = "average")
+clusters <- cutree(clustering, k = 3)
+colData(tse)$cluster <- as.factor(clusters)
+
```

Comment:
Is it necessary to show how to do this with base R as we have the clusterRows (and addCluster() ) function?

Reply:
I get what you mean, I thought we should show every basic step since it was a new exercise. I shall make the corrections, thank you :)

Reply:
Yes here we want to teach how to use mia tools so we demonstrate those.

By the way, here line 1217 is not possible: the output from transformAssay is a SE object and I don't think that it accepts as.matrix() command. See for instance:

`as.matrix(GlobalPatterns)`

> Error in h(simpleError(msg, call)) : 
>   error in evaluating the argument 'x' in selecting a method for function 'as.matrix': no method for coercing this S4 class to a vector
> 


-> Did you try the code before pushing the PR?